### PR TITLE
CDCG-90 Update service in patient module

### PR DESCRIPTION
**Pull request by ThayliAreliVillegas112**
### Feature / Bug Description
Se mostraban todos los pacientes cuando era innecesario lo que ocasionaba mayor trafico de datos

### Solution
Se implementó un nuevo método para consumir el nuevo servicio con el cual solo se mandan a llamar los pacientes del estado a filtrar  haciendo más eficiente el sistema.

### Notes
Add some notes

### Tickets
[CDCG-90
getPatientsDisabled
patientsactivedisabled
](https://cdentalcaregroup.atlassian.net/browse/)

### Screenshots / Screencasts
Provide screenshots or screencasts demoing your changes. 200px width is a good default for most screenshots.
e.g. screenshot

### CODE REVIEW

**Review by imadiaz — APPROVED** [review]

File: src/pages/patients/Patients.tsx [hunk 1]
```diff
@@ -45,14 +46,47 @@ const Patients = (props: PatientsProps) => {
...
+        try {
+            setIsLoading(true);
+            const statusList = Object.entries(filterStatus)
+                .filter(([key, value]) => value)
+                .map(([key]) => key);
+    
+            const data = await Promise.all(
+                statusList.map(async (status) => {
+                    const result = await getPatientsByStatusMutation({ status });
+                    return result as { data: Patient[] };
```
Comment (imadiaz) [inline src/pages/patients/Patients.tsx]: just a question is this cast necessary? if in the service you already set the type of data you don't need to add it again here

File: src/services/patientService.ts [hunk 2]
```diff
@@ -6,6 +6,15 @@ import { apiSlice } from "./apiSlice";
 
 export const patientService = apiSlice.injectEndpoints({
     endpoints: (builder) => ({
+      getPatientsByStatus: builder.mutation<Patient[], any>({
+        query: (data) => ({
+          url: '/patient/getPatient/filter-by-status',  // Ajusta la URL según tu implementación
```
Comment (imadiaz) [inline src/services/patientService.ts]: same here you don't need to add the comment

**Comment by ThayliAreliVillegas112** [comment]
@imadiaz I already corrected the file

File: src/pages/patients/Patients.tsx [hunk 3]
```diff
@@ -61,6 +61,36 @@ const Patients = (props: PatientsProps) => {
...
+                .filter(([key, value]) => value)
+                .map(([key]) => key);
+
+            const data = await Promise.all(
+                statusList.map(async (status) => {
+                    const result = await getPatientsByStatusMutation({ status });
+                    return result;
+            })
+            );
+            const filterData = data.flatMap((result) => 'data' in result ? result.data : []);
```
Comment (imadiaz) [inline src/pages/patients/Patients.tsx]: why do you need to add this 'data'? if you cast the data from the api it doesn;t work?

**Comment by imadiaz** [comment]
> @imadiaz I already corrected the file

@ThayliAreliVillegas112  please check the comments above

**Comment by ThayliAreliVillegas112** [comment]
@imadiaz  It's true, it wasn't necessary to do that.

fixBug2

File: src/services/patientService.ts (hunk 2 above)
Comment (imadiaz) [reply src/services/patientService.ts]: just please check this one @ThayliAreliVillegas112  and then you can go ahead with the merge

Comment (ThayliAreliVillegas112) [reply src/services/patientService.ts]: @imadiaz ok, thank you

**Review by IsaacGalindoBarbina — APPROVED** [review]

**Comment by imadiaz** [comment]
> @imadiaz It's true, it wasn't necessary to do that.
> 
> fixBug2

Go ahead with the merge

**Pull request merged**